### ISSUE TITLE
[React Recorder] Vp8 codec for video with audio

### ISSUE THREAD

**Issue by lucailvec** [Ziggeo/ziggeo-client-sdk#30]
Hello everyone,
there is a problem with the getUserMedia API and codecs used by Ziggeo React component so under Firefox the React component doesn't work.

## The problem

Under Firefox 71.0 with Fedora 30 we wrote a component that render the <ZiggeoReact/> but only after acquired access from audio and video. So by doing that and _how to reproduce_  we get this error:

```bash
An audio track cannot be recorded: video/webm;codecs=vp8 indicates an unsupported codec
```

But on the same browser we done some test. So the vp8 codec is available and by executing:

```javascript
>>> MediaRecorder.isTypeSupported("video/webm;codecs=vp8")
true
>>> MediaRecorder.isTypeSupported("video/webm;codecs=vp8,opus")
true
>>> MediaRecorder.isTypeSupported("video/webm;codecs=vp8,vorbis")
false
```

We check the availability of the recording capability of the browser right?

## How to reproduce

We use:
 - React recorder API;
 - access to private a function of the recorder;
 - Firefox 71.

```javascript
import { ZiggeoRecorder } from "react-ziggeo";
....
//react component
function functionalComponent(props){
	...
    //some logic
    ...
    render <ZiggeoRecorder
          timeminlimit={60}
          timelimit={120}
          countdown={10}
          rerecordable={false}
          picksnapshots={false}
          skipinitial
          apiKey={API_KEY}
          flip-camera={true}
          autorecord
          gallerysnapshots={0}//to avoid the selection phase before uploading
          onCountdown={() => {
          //connected on the issue ...
            if (recorderRef.current.recorder.recorder._recorder._options.videoId !== videoDeviceId)
              recorderRef.current.recorder.recorder._recorder.selectCamera(
                videoDeviceId
              );
          }}
          onRecording={...}
          onProcessed={...}
          onEnded={...}
          onProcessing={...}
          onUploadProgress={...}
          onAttached={...}
          onRef={...}
        /> ;
}
```

## The codec vp8 and the presence or not of audio in the stream requested

Another test was made not on ZiggeoRecorder but on a custom recorder component made by wrapping the MediaRecorder api.
When a MediaRecorder is created from a stream _without_ the audio (so by retrive a `...getUserMedia({video: true, audio: false})`), by specifing the codec vp8 in the MediaRecorder the recording works well. As expected (i don't know why but the problem is reproducible) with a stream created from `...getUserMedia({video: true, audio: true})` the MediaRecorder fail as Ziggeo recorder fail.

## Question

 1. What is the proper method to set a specific deviceId for **video** and **audio** in the javascript recorder?
 2. Is there a method to set a different codec?
 
Thank you

**Comment by sambua** [comment]
Hello @lucailvec,

1. By default ZiggeoRecorder will list all possible audio/video inputs, so you can switch between them. 
2. Unfortunately we don't have such method, but recorder do set best suitable codec supported by browsers.

Thanks.

**Comment by JonathanBristow** [comment]
Hi @sambua 

I am also seeing this issue.
The demo on your site works but when I have it setup in mine it doesn't. I only have 1 camera and 1 mic..

Any advice?

**Comment by ntshcalleia** [comment]
Hi @JonathanBristow,

Which version of the sdk are you using? I think this was fixed on 2.34.14.
